challenge-manager/challenge-tree/inherited_timer.go

```diff
@@ -46,10 +46,11 @@ func (ht *RoyalChallengeTree) UpdateInheritedTimer(
 	if isOneStepProven(ctx, edge, status) {
 		return math.MaxUint64, nil
 	}
-	inheritedTimer, err := ht.inheritedTimerForEdge(edge, blockNum)
+	localTimer, err := ht.LocalTimer(edge, blockNum)
 	if err != nil {
 		return 0, err
 	}
+	inheritedTimer := protocol.InheritedTimer(localTimer)
```

Comment:
if block challenge edge and has claim, we should also add the assertion unrivaled blocks here

Reply:
done.